plugins/nginx-vhosts/functions

```diff
@@ -251,13 +204,18 @@ EOF
       validate_nginx && restart_nginx
     fi
 
-    if [[ "$(is_app_vhost_enabled $APP)" == "true" ]]; then
+    if ([[ -n "$NONSSL_VHOSTS" ]] || [[ -n "$SSL_VHOSTS" ]]) && [[ "$IS_APP_VHOST_ENABLED" == "true" ]]; then
       echo "# THIS FILE IS GENERATED BY DOKKU - DO NOT EDIT, YOUR CHANGES WILL BE OVERWRITTEN" > $URLS_PATH
-      xargs -i echo "https://{}" <<< "${SSL_VHOSTS}" >> $URLS_PATH
-      xargs -i echo "http://{}" <<< "${NONSSL_VHOSTS}" >> $URLS_PATH
+      if [[ -n "$RAW_TCP_PORTS" ]]; then
+        for p in $RAW_TCP_PORTS; do
+          xargs -i echo "$SCHEME://{}:${p}" <<< "$(echo ${SSL_VHOSTS} ${NONSSL_VHOSTS} | tr ' ' '\n' | sort -u)" >> $URLS_PATH
```

Comment:
Too bad we can't include `:` in the port, then we could just set RAW_TCP_PORTS to an empty string and consolidate this for with the else below.